llvm/lib/Transforms/IPO/ArgumentPromotion.cpp

```diff
@@ -485,11 +485,39 @@ static bool allCallersPassValidPointerForArgument(
   });
 }
 
+// Try to prove that all Calls to F do not modify the memory pointed to by Arg,
+// using alias analysis local to each caller of F.
+static bool isArgUnmodifiedByAllCalls(Function *F, unsigned ArgNo,
+                                      FunctionAnalysisManager &FAM) {
```

Comment:
Can we pass just in an `Argument*` here and use `Argument->getParent()` and `Argument->getArgNo()` to simplify the function signature?

Reply:
Good point, that's much cleaner - I've made this change in commit [Address review comments 2](https://github.com/llvm/llvm-project/pull/106216/commits/aa809e9dc8a9bf5b5fc0cdc6f56549bd7806879d)